### ISSUE TITLE
Details about training super resolution model

### ISSUE THREAD

**Issue by GioFic95** [CompVis/latent-diffusion#10]
Hi @rromb, @ablattmann, @pesser, and thank you for making your great work publicly available.

Could you please supply the code for the class `ldm.data.openimages.SuperresOpenImagesAdvancedTrain`/`Validation` to train your model for super-resolution, as required in `bsr_sr/config.yaml` (see [this line](https://github.com/CompVis/latent-diffusion/blob/main/models/ldm/bsr_sr/config.yaml#L64))?
Otherwise, some more information about how to train the SR model with datasets not included in your repository would be very helpful.

Thank you very much!

**Comment by roimulia2** [comment]
> Hi @roimulia2, yes, the link of the pre-trained LDM for super-resolution is this one: [https://ommer-lab.com/files/latent-diffusion/sr_bsr.zip](https://ommer-lab.com/files/latent-diffusion/sr_bsr.zip).
> 
> You can find it in this table in the readme, looking for the task "Super-resolution": [https://github.com/CompVis/latent-diffusion#pretrained-ldms](https://github.com/CompVis/latent-diffusion#pretrained-ldms).

Sorry! I meant for the inpainting pre-trained models. Is it available as well?

**Comment by roimulia2** [comment]
@GioFic95 replied above

**Comment by GioFic95** [comment]
@roimulia2 in the ["Inpainting" section of the readme](https://github.com/CompVis/latent-diffusion#inpainting) they provide a command and the link for the pretrained models for inpainting too.

**Comment by roimulia2** [comment]
Does this make sense that the weight size is 3.1GB?

**Comment by kaihe** [comment]
@GioFic95 refer to [this line](      target: torch.nn.Identity).  second_stage_model of SR ddpm has no encode function,  therefore cond_stage_key image is still in **Image Space** not **Latent Space**.  Hence this line 
`        elif self.conditioning_key == 'concat':
            xc = torch.cat([x] + c_concat, dim=1)`
will throw _Sizes of tensors must match except in dimension 2_

**Comment by kaihe** [comment]
> @GioFic95 refer to [this line](https://github.com/CompVis/latent-diffusion/blob/main/models/ldm/bsr_sr/config.yaml#L56). second_stage_model of SR ddpm has no encode function, therefore cond_stage_key image is still in **Image Space** not **Latent Space**. Hence this line ` elif self.conditioning_key == 'concat': xc = torch.cat([x] + c_concat, dim=1)` will throw _Sizes of tensors must match except in dimension 2_
> 
> Any chance bsr_sr/config.yaml is wrong?

I finally figured it out the config is right, according to section 4.4 of the paper, 
 _and simply concatenate the low-resolution conditioning y and the inputs to the UNet, i.e. τθ is the identity._
low resolution image show be **exact the same size** with latent space,  for example a 64x64x3 kl encoder is only able to upgrade  64x64 image; and a 32x32x4 kl encoder is only able to upgrade 32x32 image.

This is very different from [SR3](https://arxiv.org/pdf/2104.07636.pdf), which will upscale whatever low resolution image to high resolution and cancat them in image space. 

I did try to upscale low resolution image, encode with second stage model and cancat them in latent space like SR3. It's always result with random noise output like this:
![image](https://user-images.githubusercontent.com/7460776/157389297-36286962-6d5c-48aa-ad21-c9643a80b790.png)

**Comment by IceClear** [comment]
@GioFic95 Hi~ Have you finally figured out where is ldm.data.openimages.SuperresOpenImagesAdvancedTrain/Validation and how to train on other datasets? I read the code pipeline and found it a little bit complicated to train on my own dataset.

**Comment by GioFic95** [comment]
@IceClear Hi, I'm in your same situation, unluckily

**Comment by IceClear** [comment]
> > @GioFic95 refer to [this line](https://github.com/CompVis/latent-diffusion/blob/main/models/ldm/bsr_sr/config.yaml#L56). second_stage_model of SR ddpm has no encode function, therefore cond_stage_key image is still in **Image Space** not **Latent Space**. Hence this line ` elif self.conditioning_key == 'concat': xc = torch.cat([x] + c_concat, dim=1)` will throw _Sizes of tensors must match except in dimension 2_
> > Any chance bsr_sr/config.yaml is wrong?
> 
> I finally figured it out the config is right, according to section 4.4 of the paper, _and simply concatenate the low-resolution conditioning y and the inputs to the UNet, i.e. τθ is the identity._ low resolution image show be **exact the same size** with latent space, for example a 64x64x3 kl encoder is only able to upgrade 64x64 image; and a 32x32x4 kl encoder is only able to upgrade 32x32 image.
> 
> This is very different from [SR3](https://arxiv.org/pdf/2104.07636.pdf), which will upscale whatever low resolution image to high resolution and cancat them in image space.
> 
> I did try to upscale low resolution image, encode with second stage model and cancat them in latent space like SR3. It's always result with random noise output like this: ![image](https://user-images.githubusercontent.com/7460776/157389297-36286962-6d5c-48aa-ad21-c9643a80b790.png)

In my view, this may not be true since I have successfully generated the 120x120 image using the pre-trained model whose size is 64x64 in the default config. I think the basic idea is that the latent code is generated based on the low-resolution input. Thus, just change the image size in the config to the desired size (must be the multiple of 8) and we can obtain SR images accordingly. But the pre-trained model can only be applied for 4x because it uses f=4, VQ. I am not sure if I am right but the generated image seems reasonable.

![bird](https://user-images.githubusercontent.com/22350795/174651431-605f79a3-3ef4-4fd0-b23b-1127865973da.png)

**Comment by andyRyuSk** [comment]
also waiting for ldm.data.openimages.SuperresOpenImagesAdvancedTrain

**Comment by jujaryu** [comment]
I need a SuperresOpenImagesAdvancedTrain too

**Comment by YunjinChen** [comment]
@GioFic95 @kaihe @IceClear hi, can you share your inference script of the LDM-BSR model? I encountered a bit problem to reproduce the bsr results shown in the paper.
